### PR TITLE
[react-relay] fix for createPaginationContainer

### PR DESCRIPTION
**Pull request by mattkrick**
- `loadMore` call signature updated to match the docs: https://facebook.github.io/relay/docs/en/pagination-container.html#loadmore

- `getConnectionFromProps` should be able to handle a null pageInfo just like the undefined that it already can (reproduce by using createPaginationContainer)

- I'm up for bumping the version to ~1.6

### CODE REVIEW

File: types/react-relay/index.d.ts [hunk 1]
```diff
@@ -131,15 +132,14 @@ export interface PageInfo {
...
+    pageInfo?: Partial<PageInfo> | null;
 }
 export type RelayPaginationProp = RelayProp & {
     hasMore(): boolean;
     isLoading(): boolean;
     loadMore(
         pageSize: number,
-        callback: (error?: Error) => void,
-        options?: RefetchOptions
+        callback?: (error?: Error) => void
```
Comment (alloy) [inline types/react-relay/index.d.ts]: The docs didn't reflect reality, so I've sent a PR to update the docs https://github.com/facebook/relay/pull/2619.

**Review by alloy — CHANGES_REQUESTED** [review]
Thanks! Can you revert the `loadMore` change?

**Comment by mattkrick** [comment]
thanks for the review @alloy!
reverted `options`, made `callback` optional/nullable so it doesn't yell at me when i write `loadMore(20)`

**Comment by alloy** [comment]
Perfect, thanks again!

**Pull request merged**